spec/features/items/index_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+require 'rails_helper'
+
+RSpec.describe "Items Index", type: :feature do
+  describe "As a visitor, when I visit the Items Index page" do
+    it "I can see a list of items with their name, price, supermarket name, and count of customers who bought that item" do
+      @supermarket_1 = Supermarket.create(name: "Reasors", location: "Tulsa, OK")
+      @supermarket_2 = Supermarket.create(name: "Whole Foods", location: "Tulsa, OK")
+      @customer_1 = Customer.create(name: "Will")
+      @customer_2 = Customer.create(name: "Dee")
+      @item_1 = @supermarket_1.items.create(name: "Banana", price: 1)
+      @item_2 = @supermarket_1.items.create(name: "Apple", price: 1)
+      @item_3 = @supermarket_2.items.create(name: "Cilantro", price: 1)
+      @item_4 = @supermarket_2.items.create(name: "Eggs", price: 2)
+      @customer_item_1 = CustomerItem.create(customer_id: @customer_1.id, item_id: @item_1.id)
+      @customer_item_2 = CustomerItem.create(customer_id: @customer_1.id, item_id: @item_2.id)
+      @customer_item_3 = CustomerItem.create(customer_id: @customer_2.id, item_id: @item_3.id)
+      @customer_item_4 = CustomerItem.create(customer_id: @customer_2.id, item_id: @item_4.id)
+
+      visit '/items'
+
+      expect(page).to have_content(@item_1.name)
+      expect(page).to have_content(@item_1.price)
+      expect(page).to have_content(@supermarket_1.name)
+      expect(page).to have_content(@item_1.customers.count)
```

Comment:
To test this logic, we don't want to call the same methods that you're calling in the production code. I know this is a simple case, but it's best practice not to use the same implementation in tests that you use in methods. Consider a more complex case where your chain of AR methods is incorrect. If you use that same combination of AR methods in the test, the test would pass despite the logic being incorrect. It would be more meaningful to use the real integer that you expect here.

Reply:
Okay so here I should have put expect(page). to have_content("Banana") or whatever I had? So im looking for the string literal and/or integer on the page?

Reply:
I'm thinking about this more for line 26 -- where you do the count. It would be more robust to put the integer you expect. For the assertions before that you're fine to call on the object attributes! It is easier to do so generally because then you don't have raw strings you have to change if you update the test data. But, when you're doing a calculation, you want to make sure you test for the result rather than including the same methods for calculating in the test